src/v/kafka/protocol/schemata/generator.py

```diff
@@ -632,6 +632,9 @@ def make_context_field(path):
     "ValueData",
```

Comment:
> Support the possibility of a tagged field that is a struct, but that
is not to be treated as an array.

can we add some more context to the commit message, and maybe an example? i'm sure there was a lot of knowledge acquired figuring this out that we should capture.